src/sentry/backup/services/import_export/impl.py

```diff
@@ -432,29 +433,24 @@ def import_by_model(
             )
 
         except DatabaseError as e:
-            # This race-detection code is a bit hacky, since it relies on string matching the error
-            # description from postgres but... ¯\_(ツ)_/¯.
-            if len(e.args) > 0:
-                desc = str(e.args[0])
-
-                # Any `UniqueViolation` indicates the possibility that we've lost a race. Check for
-                # this explicitly by seeing if an `ImportChunk` with a matching unique signature has
-                # been written to the database already.
-                if desc.startswith("UniqueViolation"):
-                    try:
-                        existing_import_chunk = get_existing_import_chunk(
-                            batch_model_name, import_flags, import_chunk_type, min_ordinal
-                        )
-                        if existing_import_chunk is not None:
-                            logger.warning("import_by_model.lost_import_race", extra=extra)
-                            return existing_import_chunk
-                    except Exception:
-                        sentry_sdk.capture_exception()
-                        return RpcImportError(
-                            kind=RpcImportErrorKind.Unknown,
-                            on=InstanceID(import_model_name),
-                            reason=f"Unknown internal error occurred: {traceback.format_exc()}",
-                        )
+            # Any `UniqueViolation` indicates the possibility that we've lost a race. Check for
+            # this explicitly by seeing if an `ImportChunk` with a matching unique signature has
+            # been written to the database already.
+            if isinstance(e.__cause__, psycopg2.errors.UniqueViolation):
```

Comment:
because we're no longer clobbering the error instance we now have direct access to the underlying postgres error \o/ (though, this broke because the original string matching now no longer works lol)